test/behaviour/typeql/TypeQLSteps.java

```diff
@@ -694,34 +695,34 @@ public boolean check(Concept concept) {
             }
 
             Set<Attribute<?>> keys = concept.asThing().asRemote(tx()).getHas(true).collect(Collectors.toSet());
-            HashMap<Label, String> keyMap = new HashMap<>();
 
             for (Attribute<?> key : keys) {
-                String keyValue;
+                if (!key.getType().getLabel().equals(type))
+                    continue;
                 switch (key.getType().getValueType()) {
                     case BOOLEAN:
-                        keyValue = key.asBoolean().getValue().toString();
-                        break;
+                        return Boolean.valueOf(value).equals(key.asBoolean().getValue());
                     case LONG:
-                        keyValue = key.asLong().getValue().toString();
-                        break;
+                        return Long.valueOf(value).equals(key.asLong().getValue());
                     case DOUBLE:
-                        keyValue = key.asDouble().getValue().toString();
-                        break;
+                        return equalsApproximate(Double.parseDouble(value), key.asDouble().getValue());
                     case STRING:
-                        keyValue = key.asString().getValue();
-                        break;
+                        return value.equals(key.asString().getValue());
                     case DATETIME:
-                        keyValue = key.asDateTime().getValue().toString();
-                        break;
+                        LocalDateTime dateTime;
+                        try {
+                            dateTime = LocalDateTime.parse(value);
+                        } catch (DateTimeParseException e) {
+                            dateTime = LocalDate.parse(value).atStartOfDay();
```

Comment:
why do we need this?

Reply:
The `DATETIME` value in the step could be written as DateTime or Date. This code snippet already existed in `ValueUniquenessCheck` method, I just copied it to consistency.